assets/react/v3/entries/course-builder/components/modals/LessonModal.tsx

```diff
@@ -372,14 +385,17 @@ const LessonModal = ({
                 render={(controllerProps) => (
                   <FormSwitch
                     {...controllerProps}
+                    disabled={!isTutorPro || !isAddonEnabled('Tutor Course Preview')}
```

Comment:
Why passing `'Tutor Course Preview'` text as function param?

Reply:
This function receives the addon name as `params`.  I have added `typescript` type `suggestion` support so the developer does not accidentally type wrong addon names.

Reply:
But shouldn't it be translatable? And it would be more readable if you put that text into some variable and pass the variable.

Reply:
No, it shouldn't be translatable. It is just the addon name that is defined in `tutorConfig.addons_data`. I have defined an `enum Addons` that can be used as `Addons.addonName`. I will replace the string with `Addons.addonName`